nemo_skills/utils.py

```diff
@@ -265,24 +270,55 @@ def chunk_data(data: List[Any], output_filename: str, chunk_id: Optional[int], n
                 f"chunk_id should be in the range [0, num_chunks)."
             )
 
-        chunk_size = len(data) // num_chunks
-        start_idx = chunk_id * chunk_size
-        end_idx = (chunk_id + 1) * chunk_size if (chunk_id < num_chunks - 1) else len(data)
+        remainder = len(data) % num_chunks
+        base_size = len(data) // num_chunks
+
+        extra = 1 if chunk_id < remainder else 0
+
+        if chunk_id < remainder:
+            start_idx = chunk_id * (base_size + 1)
+        else:
+            start_idx = remainder * (base_size + 1) + (chunk_id - remainder) * base_size
```

Comment:
Why is this logic necessary vs the earlier one?

Reply:
In your previous way, the last chunk might be much bigger than previous chunks, e.g. if `len(data) % num_chunks = num_chunks - 1` and num_chunks is sufficiently big. But in the new way, we would always distribute remainder uniformly across all chunks, thus keeping the distribution more balanced